src/main/java/de/progen_bot/listeners/StarBoardListener.java

```diff
@@ -0,0 +1,75 @@
+package de.progen_bot.listeners;
+
+import java.awt.Color;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import de.progen_bot.core.Main;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.events.message.guild.react.GuildMessageReactionAddEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+
+public class StarBoardListener extends ListenerAdapter {
+    private final Map<String, Message> starMessage = new HashMap<>();
+    private final Map<String, Integer> starCount = new HashMap<>();
+
+    final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+
+    public StarBoardListener() {
+        scheduler.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                starMessage.forEach((k, v) -> {
+                    printResult(k, v, starCount.get(k));
+                });
+                starMessage.clear();
+                starCount.clear();
+            }
+        }, 0, 1, TimeUnit.DAYS);
+    }
+
+    @Override
+    public void onGuildMessageReactionAdd(GuildMessageReactionAddEvent event) {
+
+        if (event.getReactionEmote().getName().equals("\u2B50")) {
+
+            Message mMessage = event.getJDA().getTextChannelById(event.getChannel().getId())
+                    .retrieveMessageById(event.getMessageId()).complete();
```

Comment:
Use `#queue()` since complete will block the main thread. Also `mMessage` can be null here.

Reply:
true is fixed in the new commit